gms/gossiper.cc

```diff
@@ -518,7 +518,8 @@ void gossiper::init_messaging_service_handler() {
         auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
         return handle_echo_msg(from, generation_number_opt);
     });
-    _messaging.register_gossip_shutdown([this] (inet_address from, rpc::optional<int64_t> generation_number_opt) {
+    _messaging.register_gossip_shutdown([this] (const rpc::client_info& cinfo, gms::inet_address, rpc::optional<int64_t> generation_number_opt) {
```

Comment:
> messaging, gossiper: pass client_info to shutdown message handler

Just to log it? Can you add a motivation to the commit message?

Reply:
Sure.  The motivation is to convey the sender's host_id in it.

Reply:
Current verb does not rely on `rpc::client_info` to get a source. The id of the node that is been shutting down is sent as one of the parameters. So you want to change that and take the info from `rpc::client_info` instead?

Reply:
But we send the endpoint `inet_address`.
I'd like to take the info from `client_info` to get the host_id from there, assuming the shutdown message is always sent from the shutting down node, not on behalf of another node.